pkg/java/jar/parse.go

```diff
@@ -201,6 +211,9 @@ func (p *Parser) parseArtifact(fileName string, size int64, r dio.ReadSeekerAt)
 			zap.String("artifact", fileProps.String()))
 		libs = append(libs, fileProps.library())
 	} else if !xerrors.Is(err, ArtifactNotFoundErr) {
+		if partialGAVInfoExist {
```

Comment:
If the error is ArtifactNotFoundErr and partialGAVInfoExists then we are not returning manifestProps.library(). How about?
```
if err == nil {
		log.Logger.Debugw("POM was determined in a heuristic way", zap.String("file", fileName),
			zap.String("artifact", fileProps.String()))
		libs = append(libs, fileProps.library())
} else if partialGAVInfoExist {
	return append(libs, manifestProps.library()), nil, nil
}
```

Reply:
We dont need this else, bcs we have checks here 
1. https://github.com/deepfactor-io/go-dep-parser/pull/1/files#diff-92579992d4a9b9a4dd9060431734a3ad8b8d0f4ffe5de3ab48d9343ed1dd0da4R200
2. https://github.com/deepfactor-io/go-dep-parser/pull/1/files#diff-92579992d4a9b9a4dd9060431734a3ad8b8d0f4ffe5de3ab48d9343ed1dd0da4R214

Reply:
The problem is that 		```if partialGAVInfoExist {``` is currently in the ```else if !xerrors.Is(err, ArtifactNotFoundErr) {``` Assume err is ArtifactNotFoundErr i.e ```p.searchByArtifactID(fileProps.artifactID)``` return ArtifactNotFoundErr , then ```else if !xerrors.Is(err, ArtifactNotFoundErr) {``` condition will be skipped and we will end up returning ```return nil, nil, xerrors.Errorf("failed to search by artifact id: %w", err)``` Am I missing something?

Reply:
done pls check